Mathlib/Probability/Distributions/Pareto.lean

```diff
@@ -0,0 +1,155 @@
+/-
+Copyright (c) 2024 Alvan Caleb Arulandu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alvan Caleb Arulandu
+-/
+import Mathlib.Probability.Notation
+import Mathlib.Probability.CDF
+import Mathlib.Analysis.SpecialFunctions.ImproperIntegrals
+
+/-! # Pareto distributions over ℝ
+
+Define the pareto measure over the reals.
+
+## Main definitions
+* `paretoPDFReal`: the function `t r x ↦ r * t ^ r * x ^ -(r + 1)`
+  for `t ≤ x` or `0` else, which is the probability density function of a pareto distribution with
+  scale `t` and shape `r` (when `ht : 0 < t ` and `hr : 0 < r`).
+* `paretoPDF`: `ℝ≥0∞`-valued pdf,
+  `paretoPDF t r = ENNReal.ofReal (paretoPDFReal t r)`.
+* `paretoMeasure`: a pareto measure on `ℝ`, parametrized by its scale `t` and shape `r`.
+* `paretoCDFReal`: the CDF given by the definition of CDF in `ProbabilityTheory.CDF` applied to the
+  pareto measure.
+-/
+
+open scoped ENNReal NNReal
+
+open MeasureTheory Real Set Filter Topology
+
+namespace ProbabilityTheory
+
+section ParetoPDF
+
+/-- The pdf of the pareto distribution depending on its scale and rate -/
+noncomputable
+def paretoPDFReal (t r x : ℝ) : ℝ :=
+  if t ≤ x then r * t ^ r * x ^ (-(r + 1)) else 0
+
+/-- The pdf of the pareto distribution, as a function valued in `ℝ≥0∞` -/
+noncomputable
+def paretoPDF (t r x : ℝ) : ℝ≥0∞ :=
+  ENNReal.ofReal (paretoPDFReal t r x)
+
+lemma paretoPDF_eq (t r x : ℝ) :
+    paretoPDF t r x = ENNReal.ofReal (if t ≤ x then r * t ^ r * x ^ (-(r + 1)) else 0) := rfl
+
+lemma paretoPDF_of_neg {t r x : ℝ} (hx : x < t) : paretoPDF t r x = 0 := by
+  simp only [paretoPDF_eq, if_neg (not_le.mpr hx), ENNReal.ofReal_zero]
+
+lemma paretoPDF_of_nonneg {t r x : ℝ} (hx : t ≤ x) :
+    paretoPDF t r x = ENNReal.ofReal (r * t ^ r * x ^ (-(r + 1))) := by
+  simp only [paretoPDF_eq, if_pos hx]
+
+/-- The Lebesgue integral of the pareto pdf over reals < t equals 0 -/
+lemma lintegral_paretoPDF_of_nonpos {t r x : ℝ} (hx : x < t) :
+    ∫⁻ y in Iio x, paretoPDF t r y = 0 := by
+  rw [setLIntegral_congr_fun (g := fun _ ↦ 0) measurableSet_Iio]
+  · rw [lintegral_zero, ← ENNReal.ofReal_zero]
+  · simp only [paretoPDF_eq, ge_iff_le, ENNReal.ofReal_eq_zero]
+    filter_upwards with a (_ : a < _)
+    rw [if_neg (by linarith)]
+
+/-- The pareto pdf is measurable. -/
+@[measurability]
+lemma measurable_paretoPDFReal (t r : ℝ) : Measurable (paretoPDFReal t r) :=
+  Measurable.ite measurableSet_Ici ((measurable_id.pow_const _).const_mul _) measurable_const
+
+/-- The pareto pdf is strongly measurable -/
+@[measurability]
+ lemma stronglyMeasurable_paretoPDFReal (t r : ℝ) :
```

Comment:
```suggestion
@[measurability, fun_prop]
 lemma stronglyMeasurable_paretoPDFReal (t r : ℝ) :
```

Reply:
I get a "unrecognized function property" when I try to do this here. the previous occurrence works though

Reply:
Ah, this will be because strong measurability is not conisdered as something that should be proved with `fun_prop` leave it as it is then.